examples/statistics/time_series_histogram.py

```diff
@@ -0,0 +1,85 @@
+"""
+=====================
+Time Series Histogram
+=====================
+
+This example demonstrates how to efficiently visualize large numbers of time
+series in a way that could potentially reveal hidden substructure and patterns
+that are not immediately obvious.
+
+The first plot shows the typical way of visualizing multiple time series by
+overlaying them on top of each other with ``plt.plot``. The second and third
+plots show how to reinterpret the data as a 2d histogram, with optional
+interpolation.
+"""
+from copy import copy
+import time
+
+import numpy as np
+import numpy.matlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+_, axes = plt.subplots(nrows=3, figsize=(10, 6 * 3))
+
+# Make some data; a 1D random walk + small fraction of sine waves
+num_series = 1000
+num_points = 100
+SNR = 0.10  # Signal to Noise Ratio
+x = np.linspace(0, 4 * np.pi, num_points)
+# random walk
+Y = np.cumsum(np.random.randn(num_series, num_points), axis=-1)
+# sinusoidal signal
+num_signal = int(round(SNR * num_series))
+phi = (np.pi / 8) * np.random.randn(num_signal, 1)  # small random offest
+Y[-num_signal:] = (
```

Comment:
I'll admit I'm not super clear what this is doing.  It would be nice if you could explain the underlying signal in the intro or here so it is clear what the reader should be looking for in the data.    Are you saying *some* signals are random walk and others are sines?  Why does the amplitude of the sine change?

Reply:
Sure, so what I'm trying to do is show that this method of plotting can be helpful to find patterns buried under some noisy background. The random walk is the "noise/background", and the sinusoid is the "signal/pattern".

I add a little bit of random noise to the sine by 1) adding a small random offset `phi` to each series to shift it a bit left/right, and 2) add a little bit of additive noise with the `np.random.randn` to shift each point up/down. I would never expect a perfect sine signal in real data, so this is just to make it a bit more "realistic".

The amplitude of the sine changes simply because it would not be very visible on the plot otherwise. For a Gaussian random walk with stddev of `σ` (in this case, `σ=1`), the average displacement from the origin after `n` steps is `σ*sqrt(n)`. So I scale the amplitude of the sine by this value so that it grows along with the random walk. Otherwise, the range of the sine would be restricted to `+1/-1`, while the random walk grows to have an average amplitude of `+10/-10` (and non-negligible probability to have amplitudes of even higher magnitude as well; most of the time this plot will produce random walk series that go as high as `+30/-30`).

Does that make sense? I can add these details to the intro for clarity.